pattern/variable/VariableRegistry.java

```diff
@@ -164,31 +173,49 @@ public Set<ThingVariable> things() {
         return set(things.values(), anonymous);
     }
 
+    public Set<ValueVariable> values() {
+        return set(values.values());
+    }
+
     public Set<Variable> variables() {
         Set<Variable> output = new HashSet<>();
         output.addAll(types.values());
         output.addAll(things.values());
+        output.addAll(values.values());
         output.addAll(anonymous);
         return unmodifiableSet(output);
     }
 
     public boolean contains(Reference reference) {
-        return things.containsKey(reference) || types.containsKey(reference);
+        return things.containsKey(reference) || types.containsKey(reference) || values.containsKey(reference);
     }
 
     public Variable get(Reference reference) {
         if (things.containsKey(reference)) return things.get(reference);
-        else return types.get(reference);
+        else if (types.containsKey(reference)) return types.get(reference);
+        else return values.get(reference);
     }
 
     public TypeVariable computeTypeIfAbsent(Reference reference, Function<Reference, TypeVariable> constructor) {
         if (things.containsKey(reference)) {
             throw TypeDBException.of(VARIABLE_CONTRADICTION, reference);
+        } else if (reference.isName() && values.containsKey(Reference.value(reference.name()))) {
+            throw TypeDBException.of(VARIABLE_NAME_CONFLICT, reference.name());
         } else return types.computeIfAbsent(reference, constructor);
     }
 
     public ThingVariable computeThingIfAbsent(Reference reference, Function<Reference, ThingVariable> constructor) {
-        if (types.containsKey(reference)) throw TypeDBException.of(VARIABLE_CONTRADICTION, reference);
-        else return things.computeIfAbsent(reference, constructor);
+        if (types.containsKey(reference)) {
+            throw TypeDBException.of(VARIABLE_CONTRADICTION, reference);
+        } else if (reference.isName() && values.containsKey(Reference.value(reference.name()))) {
+            throw TypeDBException.of(VARIABLE_NAME_CONFLICT, reference.name());
+        } else return things.computeIfAbsent(reference, constructor);
+    }
+
+    public ValueVariable computeValueIfAbsent(Reference reference, Function<Reference, ValueVariable> constructor) {
+        Reference clashingConceptRef = Reference.name(reference.name());
```

Comment:
this function should take a `Reference.Value` and then the clashing reference should be `Reference asNameReference = Reference.name(reference.name())`.

Right now it's not clear why you'd unwrap a reference's name only to wrap it in a name again... the fact it must be a `Value` reference is not clear!